src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -86,8 +93,8 @@ public static ByteStringContext<ByteStringMemoryCache>.InternalScope GetSliceFro
                 var ch = id[i];
                 if (ch > 127) // not ASCII, use slower mode
                 {
-                    strLength = ReadFromUnicodeKey(id, buffer, maxStrSize, separator);
-                    goto Finish;
+                    strLength = ReadFromUnicodeKey(id, buffer, maxStrSize);
+                    break;
```

Comment:
So previously we were skipping `FindEscapedPositionsAndEscapeControls()` call that we have below, right? So effectively we didn't escape it (while we did it for non lowered ID). Is that correct?